src/order/heyting/hom.lean

```diff
@@ -0,0 +1,405 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.hom.complete_lattice
+
+/-!
+# Heyting algebra morphisms
+
+A Heyting homomorphism between two Heyting algebras is a bounded lattice homomorphism that preserves
+Heyting implication.
+
+We use the `fun_like` design, so each type of morphisms has a companion typeclass which is meant to
+be satisfied by itself and all stricter types.
+
+## Types of morphisms
+
+* `heyting_hom`: Heyting homomorphisms.
+* `coheyting_hom`: Co-Heyting homomorphisms.
+* `biheyting_hom`: Bi-Heyting homomorphisms.
+
+## Typeclasses
+
+* `heyting_hom_class`
+* `coheyting_hom_class`
+* `biheyting_hom_class`
+-/
+
+open function
+
+variables {F α β γ δ : Type*}
+
+/-- The type of Heyting homomorphisms from `α` to `β`. Bounded lattice homomorphisms that preserve
```

Comment:
Your definition doesn't seem to require that the top is preserved, so is this really a bounded lattice hom?

Reply:
Yes, see `heyting_hom_class.to_bounded_lattice_hom_class`.